VideoLocker/src/main/java/org/edx/mobile/module/notification/EdxParsePushBroadcastReceiver.java

```diff
@@ -1,132 +1,126 @@
-                return;
-            if( enableFlyinWindow ) {
-                Intent intent2 = new Intent(context, NotificationFlyinService.class);
-                intent2.putExtra("com.parse.Data", intent.getExtras().getString("com.parse.Data"));
-                context.startService(intent2);
-            }
-        }catch (Exception ex){
-            logger.debug(ex.toString());
-        }
-        super.onPushReceive(context, intent);
-    }
 
+    private static ISegment getSegment(Context context) {
+        return ((MainApplication) context.getApplicationContext()).getInjector().getInstance(ISegment.class);
+    }
 
-    protected void onPushDismiss(android.content.Context context, android.content.Intent intent) {
-        super.onPushDismiss(context, intent);
+    private static Router getRouter(Context context) {
+        return ((MainApplication) context.getApplicationContext()).getInjector().getInstance(Router.class);
     }
 
-    protected void onPushOpen(android.content.Context context, android.content.Intent intent) {
-        super.onPushOpen(context, intent);
-        CourseUpdateNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
-         String courseId = payload == null ? "" : payload.getCourseId();
-        EventBus.getDefault().postSticky(new CourseAnnouncementEvent(
-            CourseAnnouncementEvent.EventType.MESSAGE_TAPPED, courseId));
+    @Override
+    protected void onPushReceive(Context context, Intent intent) {
+        BaseNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
+        if (payload != null && ParseHandleHelper.hasNotificationHash(context, payload.getIdentifier()))
+            return;
+        super.onPushReceive(context, intent);
     }
 
-    protected java.lang.Class<? extends android.app.Activity> getActivity(android.content.Context context, android.content.Intent intent) {
-        CourseUpdateNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
-        return ParseHandleHelper.getActivityClass(payload);
+    @Override
+    protected void onPushOpen(Context context, Intent intent) {
+        ParseAnalytics.trackAppOpenedInBackground(intent);
+
+        final BaseNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
+        if (null != payload && null != payload.getAction()) {
+            switch (payload.getAction()) {
+                case ParseHandleHelper.COURSE_ANNOUNCEMENT_ACTION: {
+                    final String courseId = ((CourseUpdateNotificationPayload) payload).getCourseId();
+                    getSegment(context).trackNotificationTapped(courseId);
+                    getRouter(context).showCourseAnnouncementFromNotification(context, courseId);
+                    return;
+                }
+            }
+        }
+
+        // Default behaviour for unknown notification types
+        getSegment(context).trackNotificationTapped(null);
```

Comment:
We are sending `null` here for unhandled notifications as opposed to an empty string in the `getNotification()` implementation. We should use one of these for consistency.